src/EditorFeatures/CSharpTest/EventHookup/EventHookupCommandHandlerTests.cs

```diff
@@ -1024,6 +1024,33 @@ private void C_MyEvent()
             testState.AssertCodeIs(expectedCode);
         }
 
+        [WpfFact, Trait(Traits.Feature, Traits.Features.EventHookup)]
+        [WorkItem(49325, "https://github.com/dotnet/roslyn/issues/49325")]
+        public async Task TestTopLevelStatements()
+        {
+            var markup = @"
+using System;
+
+AppDomain.CurrentDomain.AssemblyResolve +$$
+";
+            using var testState = EventHookupTestState.CreateTestState(markup);
+            testState.SendTypeChar('=');
+            testState.SendTab();
+            await testState.WaitForAsynchronousOperationsAsync();
+
+            var expectedCode = @"
+using System;
+
+AppDomain.CurrentDomain.AssemblyResolve += this.CurrentDomain_AssemblyResolve;
```

Comment:
There is an extra unneeded `this` here. But I think this can be addressed separately.